pkg/clustermanager/cluster_manager.go

```diff
@@ -289,6 +302,23 @@ func (c *ClusterManager) UpgradeCluster(ctx context.Context, managementCluster,
 		logger.V(3).Info("External etcd is ready")
 	}
 
+	if features.IsActive(features.TaintsSupport()) {
```

Comment:
We should try to have only one place where we check this
I think moving it to a validation like I suggested might help